driver-core/src/main/com/mongodb/operation/CreateIndexesOperation.java

```diff
@@ -177,7 +177,7 @@ private BsonDocument getIndex(final IndexRequest request) {
         if (request.isUnique()) {
             index.append("unique", BsonBoolean.TRUE);
         }
-        if (request.isSparse()) {
+        if (request.isSparse() && request.getPartialFilterExpression() == null) {
```

Comment:
Silently ignoring sparse when there's a partial filter expression is problematic.  Does the server error if you set both?  If so, the driver should treat it as an error as well.  It would be fine to just let the server report the error rather than detect it client side.